backend/ringv2/ringv2.go

```diff
@@ -355,12 +363,22 @@ func (r *Ring) startWatchers(ctx context.Context, ch chan Event, name string, va
 	r.mu.Lock()
 	r.watchers[watcher.watcherKey] = watcher
 	r.mu.Unlock()
-	if err := watcher.ensureActiveTrigger(ctx); err != nil {
-		notifyError(ch, fmt.Errorf("error while starting ring watcher: %s", err))
-		notifyClosing(ch)
-		cancel()
-		return
+
+	backoff := retry.ExponentialBackoff{
```

Comment:
I think this should go in the `ensureActiveTrigger` method itself - the method is called from two different places.